src/formatDate.js

```diff
@@ -9,6 +9,62 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const outputNeeded = {
+    day: '',
+    month: '',
+    year: '',
+    separator: '',
+  };
+  const separatedDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  // check inData and start to build object for the result creating
+  for (let i = 0; i < fromFormat.length; i++) {
+    const item = fromFormat[i];
+
+    switch (item[0].toLowerCase()) {
+      case 'd':
+        outputNeeded.day = `${separatedDate[i]}`;
+        break;
+      case 'm':
+        outputNeeded.month = `${separatedDate[i]}`;
+        break;
+      case 'y':
+        const yearIn =
+          item.length === 2 && Number(separatedDate[i]) < 30
+            ? `20${separatedDate[i].slice(-2)}`
+            : `19${separatedDate[i].slice(-2)}`;
+
+        outputNeeded.year = item.length === 2 ? yearIn : `${separatedDate[i]}`;
+        break;
+      default:
+        outputNeeded.separator = `${toFormat[i]}`;
+    }
```

Comment:
The logic for determining the year when converting from 'YY' to 'YYYY' is incorrect. The condition `item.length === 2 && Number(separatedDate[i]) < 30` will always prepend '20' to the year, which is not correct for years 30 and above. Instead, you should check if the year is less than '30' to prepend '20', or prepend '19' otherwise.